### ISSUE TITLE
[Suggestion] Instruction Fine-Tuning - SFT Module

### ISSUE THREAD

**Issue by ArlindKadra** [huggingface/smol-course#60]
Thanks for taking the time into developing this interesting course. I wanted to suggest that regarding the SFT module in Chapter 1, the `bigcode/the-stack-smol` dataset seems to break the flow a bit.  Since it is not an instruction-tuning dataset, but more a domain-specific dataset with no instruction following. As such, it does not have the question/answering pairs. 

Based on that, if you train on the dataset, the response to the prompt is the same as before. Maybe switching it up to the `openai/gsm8k` dataset, or something similar? That way one would still have to prepare the dataset before feeding it to the SFTTrainer.

**Comment by burtenshaw** [comment]
That's a nice suggestion. Are you interested in opening a PR?

**Comment by ArlindKadra** [comment]
I can give it a try, however, I am not able to do it in a timely manner. I will be available starting from the 18th.

**Comment by burtenshaw** [comment]
No worries @ArlindKadra . Come back when you're ready and check if it still needs doing.

Thanks for the issue.

**Comment by asvskartheek** [comment]
Wanted to add few of my queries here, instead of creating a new issue. (I hope this is ok)

1. What is the task the model is being trained on? 
- My guess is Casual LM with cross entropy loss.

2. If it is Casual LM then are we in anyway also ignoring special characters and phrases like <|im_start|>?

3. The everyday conversations dataset has several columns, where are we specifying to train on the "chatml" templated messages column of the dataset?

Tried to get answers for these queries in SFT couldn't get answers. Maybe adding bit more detail in "The Finetuning Process" sub-section of the Supervised Fine tuning page would be helpful.

**Comment by PrabhuRajendhran** [comment]
_The following suggestions might be helpful 💯:

1. The current dataset is not an instruction dataset, meaning it does not contain question/answer pairs.

2. Use the 'content' field values to infer question/answer pairs. Specifically, use the Docstring as the Question and the Content as the Answer.

3. Convert to chat format as shown below:

```python
messages = [
    {'content': question, 'role': 'user'},
    {'content': answer, 'role': 'assistant'}
]
```

4. Apply `apply_chat_template` to the list of your messages.

5. Continue with `SFTTrainer` for fine-tuning._
